sdk/identity/azure-identity/azure/identity/_credentials/browser.py

```diff
@@ -65,21 +72,49 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
         :raises ~azure.core.exceptions.ClientAuthenticationError: authentication failed. The error's ``message``
           attribute gives a reason. Any error response from Azure Active Directory is available as the error's
           ``response`` attribute.
+        :raises ~azure.identity.AuthenticationRequiredError: the credential is configured to authenticate only silently
+          (without user interaction), and was unable to do so.
         """
-        return self._get_token_from_cache(scopes, **kwargs) or self._get_token_by_auth_code(scopes, **kwargs)
+        token = self._acquire_token_silent(*scopes, **kwargs)
+        if not token:
+            if self._silent_auth_only:
+                raise AuthenticationRequiredError()
 
-    def _get_token_from_cache(self, scopes, **kwargs):
-        """if the user has already signed in, we can redeem a refresh token for a new access token"""
-        app = self._get_app()
-        accounts = app.get_accounts()
-        if accounts:  # => user has already authenticated
-            # MSAL asserts scopes is a list
-            scopes = list(scopes)  # type: ignore
             now = int(time.time())
-            token = app.acquire_token_silent(scopes, account=accounts[0], **kwargs)
-            if token and "access_token" in token and "expires_in" in token:
-                return AccessToken(token["access_token"], now + int(token["expires_in"]))
-        return None
+            response = self._get_token_by_auth_code(scopes, **kwargs)
+            if not self._profile:
+                self._profile = _build_auth_profile(response)
+            token = AccessToken(response["access_token"], now + int(response["expires_in"]))
+
+        return token
+
+    @classmethod
+    def authenticate(cls, client_id, **kwargs):
```

Comment:
Maybe we can let `client_id` default to `None`? Thus we can automatically use `AZURE_CLI_CLIENT_ID`. Otherwise,

```py
            credential, profile = InteractiveBrowserCredential.authenticate(
                silent_auth_only=True,
                scope='https://management.azure.com/.default'
            )
```
gives
```
TypeError: authenticate() missing 1 required positional argument: 'client_id'
```

Reply:
This API is intended for applications beyond the CLI, which we don't want unintentionally authenticating users to the CLI. It's true `InteractiveBrowserCredential.get_token` does that today, but that was designed for a different scenario and in a future release its implicit default client id will change.

Reply:
I see. Thanks for the clarification.